lib/streamlit/runtime/runtime.py

```diff
@@ -166,13 +170,14 @@ def __init__(self, config: RuntimeConfig):
         self._message_cache = ForwardMsgCache()
         self._uploaded_file_mgr = UploadedFileManager()
         self._uploaded_file_mgr.on_files_updated.connect(self._on_files_updated)
+        self._media_file_manager = MediaFileManager(storage=config.media_file_storage)
+        set_media_file_manager(self._media_file_manager)
 
         self._stats_mgr = StatsManager()
         self._stats_mgr.register_provider(get_memo_stats_provider())
         self._stats_mgr.register_provider(get_singleton_stats_provider())
         self._stats_mgr.register_provider(_mem_caches)
         self._stats_mgr.register_provider(self._message_cache)
-        self._stats_mgr.register_provider(media_file_manager)
```

Comment:
MediaFileManager is no longer a CacheStatsProvider. (But the MemoryMediaFileStorage class is - server instantiates this, and therefore it's server's job to add it to stats_mgr.)